cloud/amazon/s3.py

```diff
@@ -591,13 +610,21 @@ def main():
 
     # Support for listing a set of keys
     if mode == 'list':
-        bucket_object = get_bucket(module, s3, bucket)
-
-        # If the bucket does not exist then bail out
-        if bucket_object is None:
-            module.fail_json(msg="Target bucket (%s) cannot be found"% bucket, failed=True)
-
-        list_keys(module, bucket_object, prefix, marker, max_keys)
+        if bucket == '/':
```

Comment:
For clarity, can you make this only happen on RGW flavors? This code is specific to RGW, and I think it convolutes what we had before since there's no (apparent) reason for special-casing buckets named `/`.